server/handlers/project_command_output_handler_test.go

```diff
@@ -0,0 +1,135 @@
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/handlers"
+	"github.com/runatlantis/atlantis/server/logging"
+	. "github.com/runatlantis/atlantis/testing"
+)
+
+func TestProjectCommandOutputHandler(t *testing.T) {
+	t.Run("Should Group by Project Info", func(t *testing.T) {
+		tempchan := make(chan *models.ProjectCmdOutputLine)
+		logger := logging.NewNoopLogger(t)
+		projectOutputHandler := handlers.NewProjectCommandOutputHandler(tempchan, logger)
+		ctx := models.ProjectCommandContext{
+			BaseRepo: models.Repo{
+				Name:  "test-repo",
+				Owner: "test-org",
+			},
+			HeadRepo: models.Repo{
+				Name:  "test-repo",
+				Owner: "test-org",
+			},
+			Pull: models.PullRequest{
+				Num:        1,
+				HeadBranch: "add-feat",
+				BaseBranch: "master",
+				Author:     "acme",
+			},
+			User: models.User{
+				Username: "acme-user",
+			},
+			Log:         logger,
+			Workspace:   "myworkspace",
+			RepoRelDir:  "mydir",
+			ProjectName: "test-project",
+		}
+		fmt.Printf("Testing Handle func in unit test")
+
+		var wg sync.WaitGroup
+
+		go func() {
+			projectOutputHandler.Handle()
+		}()
+
+		go func() {
+			projectOutputHandler.Send(ctx, "Test Terraform Output")
+		}()
+
+		expectMsg := ""
+		wg.Add(1)
+		go func() {
+			projectOutputHandler.Receive(ctx.PullInfo(), func(msg string) error {
+				expectMsg = msg
+				wg.Done()
+				return nil
+			})
+		}()
+		wg.Wait()
+		Equals(t, expectMsg, "Test Terraform Output")
+
+		time.Sleep(1 * time.Second)
+
+	})
+}
+
+func TestProjectCommandOutputHandler_ClearBuff(t *testing.T) {
+	t.Run("Should Group by Project Info", func(t *testing.T) {
+		tempchan := make(chan *models.ProjectCmdOutputLine)
+		logger := logging.NewNoopLogger(t)
+		projectOutputHandler := handlers.NewProjectCommandOutputHandler(tempchan, logger)
+		ctx := models.ProjectCommandContext{
+			BaseRepo: models.Repo{
+				Name:  "test-repo",
+				Owner: "test-org",
+			},
+			HeadRepo: models.Repo{
+				Name:  "test-repo",
+				Owner: "test-org",
+			},
+			Pull: models.PullRequest{
+				Num:        1,
+				HeadBranch: "add-feat",
+				BaseBranch: "master",
+				Author:     "acme",
+			},
+			User: models.User{
+				Username: "acme-user",
+			},
+			Log:         logger,
+			Workspace:   "myworkspace",
+			RepoRelDir:  "mydir",
+			ProjectName: "test-project",
+		}
+		fmt.Printf("Testing Handle func in unit test")
+
+		var wg sync.WaitGroup
+
+		go func() {
+			projectOutputHandler.Handle()
+		}()
+
+		go func() {
+			projectOutputHandler.Send(ctx, "Test Terraform Output")
+		}()
+
+		go func() {
+			projectOutputHandler.Clear(ctx, "")
+		}()
+
+		go func() {
+			projectOutputHandler.Send(ctx, "Test Terraform Output")
+		}()
+
+		expectMsg := ""
+		wg.Add(1)
+		go func() {
+			projectOutputHandler.Receive(ctx.PullInfo(), func(msg string) error {
+				expectMsg = msg
+				return nil
+			})
+			wg.Done()
+		}()
+		wg.Wait()
```

Comment:
This should fix the test
```suggestion
		wg.Add(1)
		go func() {
			projectOutputHandler.Send(ctx, "Test Terraform Output")
		}()

		wg.Add(1)
		go func() {
			projectOutputHandler.Clear(ctx, "")
		}()

		wg.Add(1)
		go func() {
			projectOutputHandler.Send(ctx, "Test Terraform Output")
		}()

		expectMsg := ""
		go func() {
			projectOutputHandler.Receive(ctx.PullInfo(), func(msg string) error {
				expectMsg = msg
				wg.Done()
				return nil
			})
		}()
		wg.Wait()
```